sky/cli.py

```diff
@@ -2675,25 +2676,26 @@ def _hint_or_raise_for_down_jobs_controller(controller_name: str):
     controller = controller_utils.Controllers.from_name(controller_name)
     assert controller is not None, controller_name
 
+    try:
+        with rich_utils.safe_status(
+                ux_utils.spinner_message('Checking controller status')):
+            handle = backend_utils.is_controller_accessible(controller,
+                                                            stopped_message='')
+    except exceptions.ClusterNotUpError as e:
+        if e.cluster_status is None:
+            click.echo('Controller has already been torn down.')
+            sys.exit(0)
+
+        spinner_msg = (f'Controller is {e.cluster_status.value} - restarting '
+                       'to check that it\'s safe to terminate.')
+        with rich_utils.safe_status(ux_utils.spinner_message(spinner_msg)):
+            with sky_logging.silent():
+                handle = sky.start(controller_name)
+
+    # Check #1: in-progress managed jobs
     with rich_utils.safe_status(
             ux_utils.spinner_message('Checking for in-progress managed jobs')):
-        try:
-            managed_jobs_ = managed_jobs.queue(refresh=False,
-                                               skip_finished=True)
-        except exceptions.ClusterNotUpError as e:
-            if controller.value.connection_error_hint in str(e):
-                with ux_utils.print_exception_no_traceback():
-                    raise exceptions.NotSupportedError(
-                        controller.value.
-                        decline_down_when_failed_to_fetch_status_hint)
-            if e.cluster_status is None:
-                click.echo(
-                    'Managed jobs controller has already been torn down.')
-                sys.exit(0)
-            # At this point, the managed jobs are failed to be fetched due to
-            # the controller being STOPPED or being firstly launched, i.e.,
-            # there is no in-prgress managed jobs.
-            managed_jobs_ = []
+        managed_jobs_ = managed_jobs.queue(refresh=False, skip_finished=True)
```

Comment:
The two steps of checking the job queue first and then checking the cluster seems not clean and can cause additional latency.

One proposal: we, instead, just add another column in managed jobs database to indicate whether a job has clusters to be cleaned up, e.g. we can add a new scheduler state for jobs, have the scheduler set the job to be scheduler state `CLEANUP` and later be set to `DONE` only after the clusters are cleaned up.

Reply:
My hypothesis is that `sky down` on a controller is very rare, and the additional latency is well worth it for the safety.

I think something like your proposal is cleaner, but even in the FAILED_CONTROLLER case most clusters will autodown and your solution will not have a way to track that. We should discuss a bit but I think it's better to patch the hole by getting this in now and following up with a better solution.